### PR TITLE
chore: fix ESLint errors 1

### PR DESCRIPTION
**Pull request by worldomonation**
#### Proposed Changes

This is a 20% time project to reduce as many code style errors/warnings as possible.

Key changes:
- refactor unit tests in several components to use Jest expect instead of assert;
- split up unit tests in `client/components/phone-input` over multiple files;
- convert individual test cases in `client/lib/phone-validation` to a parametrized test;
- disable overly restrictive `wpcalypso/jsx-classname-namespace` rule for example files. See p4TIVU-9Ok-p2#comment-11203 for some context on this rule.

#### Testing Instructions

Ensure the following:
  - [x] Calypso E2E
  - [x] Check Code Style
  - [x] Unit Tests

Related to #56330

### CODE REVIEW

**Comment by worldomonation** [comment]
@dpasque Tagging you before the PR is in fully review ready state as a heads up of what's coming.

File: client/blocks/conversation-caterpillar/docs/example.jsx
Comment (worldomonation) [inline client/blocks/conversation-caterpillar/docs/example.jsx]: As noted in the PR comment, the usefulness of this rule is discussed in p4TIVU-9Ok-p2#comment-11203.

File: client/components/popover-menu/docs/example.jsx
Comment (worldomonation) [inline client/components/popover-menu/docs/example.jsx:46]: Instead of disabling the rule I opted to fix it here.

File: docs/guide/hello-world.md
Comment (worldomonation) [inline docs/guide/hello-world.md:104]: ESLint says there's some sort of unexpected token here, but there isn't.

I've checked whether it was caused by a zero-width character, but it doesn't appear to be the case either.

In any case, removing syntax highlighting solved the problem.

File: client/tsconfig.json
Comment (worldomonation) [inline client/tsconfig.json]: Without adding this, imports within `client` that did something like:

```
import {xyz} from 'calypso/...';
```

would throw a warning about an import (which I am having a hard time locating what the issue was about).

File: package.json
Comment (worldomonation) [inline package.json:189]: Error about a missing package.

File: client/my-sites/checkout/checkout-thank-you/index.jsx
Comment (worldomonation) [inline client/my-sites/checkout/checkout-thank-you/index.jsx:16]: @Automattic/shilling This fixes the `'isJetpackBusinessPlan' is defined but never used. (no-unused-vars)` error.

File: client/blocks/conversation-caterpillar/docs/example.jsx
Comment (dpasque) [reply client/blocks/conversation-caterpillar/docs/example.jsx]: So this is maybe a stupid suggestion, but I'm going for it. 😁 But what about just fixing the rule, as per that comment? That may be wayyyy out of scope for this PR, but the way I see it is that each one of these disable comments is technically a form of technical debt, and you could knock out a lot just by patching the rule!

And it doesn't seem too hard -- from what I can tell, it would just be tweaking the Regex test here: https://github.com/Automattic/wp-calypso/blob/trunk/packages/eslint-plugin-wpcalypso/lib/rules/jsx-classname-namespace.js#L72-L86

File: client/components/file-picker/docs/example.jsx
Comment (dpasque) [inline client/components/file-picker/docs/example.jsx:51]: Could you explain this rule fix a little more?

File: docs/guide/hello-world.md
Comment (dpasque) [reply docs/guide/hello-world.md:104]: >ESLint says there's some sort of unexpected token here, but there isn't.

I think what's going on here is the JavaScript interpreter is trying to interpret this block as valid JavaScript, but it's not (because it's just an object -- there's no command, etc.) The "unexpected token" is the starting `{`.

All that to say... I think removing the syntax highlighting is a great call! Because it's not actually JavaScript! 👍

**Comment by dpasque** [comment]
@worldomonation this is such an awesome 20% time project! 🎉  This will make such a difference to finally get the code style running clean! 😄

I would recommend also having the @Automattic/team-calypso-platform  folks take a look at these too. 👍 

I left a few small comments just as I was looking over the work so far, but let me know when all the changes are done and you're ready for review!

File: client/blocks/conversation-caterpillar/docs/example.jsx
Comment (dpasque) [reply client/blocks/conversation-caterpillar/docs/example.jsx]: Again, feel free to just toss out that suggestion. 😄  But I had the idea, so I thought I'd throw it out!

Comment (tyxla) [reply client/blocks/conversation-caterpillar/docs/example.jsx]: I'm totally up for removing this rule completely FWIW. It's also being suggested in p9oQ9f-1fu-p2. I also agree with @dpasque that it's probably better if we do it in another PR.

File: client/my-sites/checkout/checkout-thank-you/index.jsx
Comment (sirbrillig) [reply client/my-sites/checkout/checkout-thank-you/index.jsx:16]: Thanks! I wonder how that got past the tests.

File: client/components/file-picker/docs/example.jsx
Comment (worldomonation) [reply client/components/file-picker/docs/example.jsx:51]: The rule that was highlighted here is:

```
The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/anchor-is-valid.md
```

Now, typically I would refer this to the team that owns this component for the fix, but since this is an example file, I felt it was okay to add a href value.

File: client/blocks/conversation-caterpillar/docs/example.jsx
Comment (worldomonation) [reply client/blocks/conversation-caterpillar/docs/example.jsx]: @dpasque @tyxla It seems like there's agreement in removing the rule, which is great! I wasn't aware of @jsnajdr's list of pre-sabbatical tasks, thanks for bringing it to my attention.

I see in the P2 that the suggestion is to replace it with a simpler rule. As that's far outside the scope of this PR, I'll revert the changes relating to the rule in this PR and we can discuss in another task/PR about actually making the changes.

**Comment by worldomonation** [comment]
@dpasque @tyxla I've removed changes to files where the only change was to disable the `wpcalypso/jsx-classname-namespace` rule. 

As discussed in https://github.com/Automattic/wp-calypso/pull/64743#discussion_r900299503 let's address the issue of this rule in another PR.

File: packages/calypso-e2e/src/lib/utils/get-test-account-by-feature.ts
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/utils/get-test-account-by-feature.ts:36]: I don't like deploying this rule, but this is a place where it makes sense.

File: client/tsconfig.json
Comment (tyxla) [reply client/tsconfig.json]: I'd like to hear more about this TBH.

File: patch.patch
Comment (tyxla) [inline patch.patch]: We need to delete this file before shipping.

File: test/e2e/specs/support/support__home.ts
Comment (tyxla) [inline test/e2e/specs/support/support__home.ts:57]: I wonder if we should just disable this rule altogether.

**Review by tyxla — APPROVED** [review]
This is pretty close IMHO. I'm actually approving it because after addressing the comments, it should be good to go. Thank you 🚀

File: client/components/file-picker/docs/example.jsx
Comment (dpasque) [reply client/components/file-picker/docs/example.jsx:51]: Ah sure, that makes sense! Linking to an element with an ID of `input` just felt like a kind of strange/uncommon application

Per that rule, it sounds like this element might be better suited as a button? It's really just there to trigger an event!

Anyway, that's a tiny nit and not required! 😄

**Review by dpasque — APPROVED** [review]
Same as Marin! With the final pending changes, LGTM! 🚀

File: patch.patch
Comment (worldomonation) [reply patch.patch]: Ah yeah, my bad...I ought to be _much more_ careful with things like this.

File: test/e2e/specs/support/support__home.ts
Comment (worldomonation) [reply test/e2e/specs/support/support__home.ts:57]: I can see the reasoning behind such rules - it avoids having skipped/disabled tests live forever without being noticed.

With that said, even with the rule enabled these tests _have been_ sitting in the purgatory for some time, so...perhaps I just proved the uselessness of the rule 😆

File: client/tsconfig.json
Comment (worldomonation) [reply client/tsconfig.json]: I initially included this for the reason cited above - it was to fix a whole bunch of import warnings for imports within `client`, but I honestly am a bit skeptical that I should do this.

I'll first remove this particular change from the PR and see if the import warnings resurface.

File: test/e2e/specs/support/support__home.ts
Comment (worldomonation) [reply test/e2e/specs/support/support__home.ts:57]: What I will do, since this PR has gone on for long enough:
1. merge this PR with the disable statements included;
2. spin up a new issue to remove or disable this ESLint rule entirely;
3. remove all instances of this rule being disabled in the repo.

I'll likely do this as my next 20% project.

Comment (worldomonation) [reply test/e2e/specs/support/support__home.ts:57]: Spun up https://github.com/Automattic/wp-calypso/issues/64870 to explore deletion of the rule.

Comment (tyxla) [reply test/e2e/specs/support/support__home.ts:57]: Nice, thanks 👍

**Pull request merged**
